src/tribler-core/tribler_core/modules/libtorrent/torrentdef.py

```diff
@@ -349,19 +348,20 @@ def _get_all_files_as_unicode_with_length(self):
                     # We assume that it is correctly encoded and use
                     # it normally
                     try:
-                        yield (join(*[ensure_unicode(element, "UTF-8") for element in file_dict[b"path.utf-8"]]),
+                        file_path_parts = (ensure_unicode(element, "UTF-8") for element in file_dict[b"path.utf-8"])
+                        yield (Path(*file_path_parts).name,
                                file_dict[b"length"])
                         continue
                     except UnicodeError:
                         pass
 
                 if b"path" in file_dict:
-                    # Try to use the 'encoding' field.  If it exists,
+                    # Try to use the 'encoding' field. If it exists,
                     # it should contain something like 'utf-8'
                     if b"encoding" in self.metainfo:
                         encoding = ensure_unicode(self.metainfo[b"encoding"], "utf8")
                         try:
-                            yield (join(*[ensure_unicode(element, encoding) for element in file_dict[b"path"]]),
+                            yield (Path(*(ensure_unicode(element, encoding) for element in file_dict[b"path"])).name,
```

Comment:
The original line does not seem to get the *last* part of the path, as you do with `.name` there.